civis/ml/_model.py

```diff
@@ -450,3 +462,518 @@ def training_metadata(self):
+    >>> train.metrics['roc_auc']
+    0.88425
+    >>> pred = model.predict(table_name='schema.demographics_table ',\
+                             database_name='My Redshift Cluster',\
+                             output_table='schema.predicted_survey_response',\
+                             if_exists='drop',\
+                             n_jobs=50)
+    >>> df_pred = pred.table  # Blocks until finished
+    # Modify the parameters of the base estimator in a default model:
+    >>> model = ModelPipeline('sparse_logistic', 'depvar',\
+                              primary_key='voterbase_id',\
+                              parameters={'C': 2})
+    # Grid search over hyperparameters in the base estimator:
+    >>> model = ModelPipeline('sparse_logistic', 'depvar',\
+                              primary_key='voterbase_id',\
+                              cross_validation_parameters={'C': [0.1, 1, 10]})
+
+    See Also
+    --------
+    ModelFuture : Subclass of :class:`~concurrent.future.Future`,
+        output by :class:`~ModelPipeline` methods
+    """
+    # These are the v1.0 templates
+    train_template_id = 8387
+    predict_template_id = 8388
+
+    def __init__(self, model, dependent_variable, *,
+                 primary_key=None, parameters=None,
+                 cross_validation_parameters=None, model_name=None,
+                 calibration=None, excluded_columns=None, client=None,
+                 cpu_requested=None, memory_requested=None,
+                 disk_requested=None, verbose=False):
+        self.model = model
+        self._input_model = model  # In case we need to modify the input
+        if isinstance(dependent_variable, str):
+            # Standardize the dependent variable as a list.
+            dependent_variable = [dependent_variable]
+        self.dependent_variable = dependent_variable
+
+        # optional but common parameters
+        self.primary_key = primary_key
+        self.parameters = parameters or {}
+        self.cv_params = cross_validation_parameters or {}
+        self.model_name = model_name  # None lets Platform use template name
+        self.excluded_columns = excluded_columns
+        self.calibration = calibration
+        self.job_resources = {'REQUIRED_CPU': cpu_requested,
+                              'REQUIRED_MEMORY': memory_requested,
+                              'REQUIRED_DISK_SPACE': disk_requested}
+        self.verbose = verbose
+
+        if client is None:
+            client = APIClient(resources='all')
+        self._client = client
+        self.train_result_ = None
+
+    def __getstate__(self) -> dict:
+        state = self.__dict__.copy()
+        del state['_client']
+        return state
+
+    def __setstate__(self, state: dict):
+        self.__dict__ = state
+        self._client = APIClient(resources='all')
+
+    @classmethod
+    def from_existing(cls, train_job_id, train_run_id='latest', client=None):
```

Comment:
Okay, I think you're right.  If we were ever to do something like an `EnsemblePipeline`, it would be nice to have the same API with `EnsemblePipeline.from_existing`.